data/sea/30-array.h

```diff
@@ -203,15 +203,16 @@ static ARRAY_T(t) INLINE ARRAY_FUN(t,put_immutable)
         ARRAY_T(t) arr = (ARRAY_T(t))imempool_alloc(pool, bytes_new);           \
         memcpy(arr, x, bytes_old);                                              \
         x = arr;                                                                \
+        x->count = ix + 1;                                                      \
                                                                                 \
     } else if (ix >= count) {                                                   \
-      ARRAY_T(t) arr = (ARRAY_T(t))imempool_alloc(pool, bytes_old);             \
-      memcpy(arr, x, bytes_old);                                                \
-      x = arr;                                                                  \
+        ARRAY_T(t) arr = (ARRAY_T(t))imempool_alloc(pool, bytes_old);           \
+        memcpy(arr, x, bytes_old);                                              \
+        x = arr;                                                                \
+        x->count = ix + 1;                                                      \
                                                                                 \
     }                                                                           \
                                                                                 \
-    x->count = ix + 1;                                                          \
     ARRAY_PAYLOAD(t,x)[ix] = v;                                                 \
```

Comment:
I don't understand what you mean. the immutable version should allocate a new array and copy the results in regardless, because at the moment it's mutating the input array if the index is already in the array